machine-learning/02-Data Preparation and Feature Engineering.ipynb

```diff
@@ -1,45 +1,405 @@
+        }
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/html": [
+              "<div><div></div><div></div><div><strong>Installed Packages</strong><ul><li><span>Microsoft.ML, 1.7.1</span></li></ul></div></div>"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "#r \"nuget: Microsoft.ML, 1.7.1\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "using Microsoft.ML;\n",
+        "using Microsoft.ML.Data;"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "#### Loading from a file\n",
+        "\n",
+        "A [TextLoader](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.textloader?view=ml-dotnet) can load a structred file into the IDataView format. Structured information is basically columns and rows of data. The loader will need to be told if the file has a header, and what separator character the file uses. \n",
+        "\n",
+        "By using the loader instead of the direct method LoadFromTextFile, there is the option to load from multiple files. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "public class ModelInput\n",
+        "{\n",
+        "    [LoadColumn(0)]\n",
+        "    [ColumnName(@\"vendor_id\")]\n",
+        "    public string Vendor_id { get; set; }\n",
+        "\n",
+        "    [LoadColumn(1)]\n",
+        "    [ColumnName(@\"rate_code\")]\n",
+        "    public float Rate_code { get; set; }\n",
+        "\n",
+        "    [LoadColumn(2)]\n",
+        "    [ColumnName(@\"passenger_count\")]\n",
+        "    public float Passenger_count { get; set; }\n",
+        "\n",
+        "    [LoadColumn(3)]\n",
+        "    [ColumnName(@\"trip_time_in_secs\")]\n",
+        "    public float Trip_time_in_secs { get; set; }\n",
+        "\n",
+        "    [LoadColumn(4)]\n",
+        "    [ColumnName(@\"trip_distance\")]\n",
+        "    public float Trip_distance { get; set; }\n",
+        "\n",
+        "    [LoadColumn(5)]\n",
+        "    [ColumnName(@\"payment_type\")]\n",
+        "    public string Payment_type { get; set; }\n",
+        "\n",
+        "    [LoadColumn(6)]\n",
+        "    [ColumnName(@\"fare_amount\")]\n",
+        "    public float Fare_amount { get; set; }\n",
+        "\n",
+        "}"
+      ]
+    },
+    {
+      "cell_type": "code",
```

Comment:
After you wrote the code to define your schema, it would be good to explain what `LoadColumn` and `ColumnName` attributes mean. I think for the most part you can copy / paste the info in this [doc](https://docs.microsoft.com/dotnet/machine-learning/how-to-guides/load-data-ml-net#annotating-the-data-model-with-column-attributes). Feel free to add your own spin / reword to make it sound better.